docs/application_logic/protocol.md

```diff
@@ -1,4 +1,46 @@
+
+1. **The legitimacy of the timestamp of the message**: Whenever a node issues a message, it adds a timestamp to it, which should represent the local time of issuance (as seen by the issuer node). The other nodes will judge if this timestamp is reasonable, by checking if it is too far away from their own local clock. The specific rule to do this check is described in LINK.
+2. In the case of a value transfer, **whether it is a conflict**: We use the **FCoB Rule** (ADD LINK). Roughly, the node will have a positive opinion about a transaction A if and only all its conflicts arrived later than a certain time interval after A's arrival. 
+
+In each round, nodes randomly choose other nodes to query about their opinions about one of the subjects above. The querying node changes its own opinion if the number of responses with a different opinion than it is greater than a certain threshold. In order to prevent liveness attacks, this threshold is determined by a random number issued by a committee of high Consensus Mana nodes via the **dRNG** application. Without the random threshold, an attacker could lie about its responses to prevent the protocol from terminating. Finally, a node will finalize on a certain opinion after holding it for a certain number of rounds. 
+
+When selecting which other nodes to query, a node must weight the list of all nodes by Consensus Mana. Thus, high Consensus Mana nodes are queried more often then low Consensus Mana nodes. This makes it difficult for an attacker to manipulate the outcome of the voting. Unless the attacker controls more than 1/3 of the Consensus Mana in the system, with high probability, we know that FPC has the following properties:
+
+1. **Termination**: every honest node will finalize on some opinion.
+2. **Agreement**: all honest nodes will finalize on the same opinion.
+3. **Integrity**: if a super majority of nodes -e.g. more than 90% weighted by Consensus Mana-, have the same initial opinion, then FPC will terminate with that value.
+
+![title](Protocol_overview_consensus.png)
+
+Some concepts or modules implemented -even not strictly being part of the protocol- have an extreme importance in the understanding of the protocol as a whole, such as the **Grades of Finality** (add link). Analogously to Bitcoin's [six blocks rule](https://en.bitcoin.it/wiki/Confirmation), our protocol also has certain measures of the probability of a certain message being considered valid permanently by all nodes. The four possible grades are:
+
+1. **Grade 1**: If 10 seconds have passed since the time the message arrived and no conflict was seen.
+2. **Grade 2**: If 20 seconds have passed since the time the message arrived and no conflict was seen.
+3. **Grade 3**: If the message achieved grade 2 and it accumulated enough approval weight (see LINK for more details)
+4. **Grade 4**: If the message achieved grade 3 and 30 minutes (??????????) have passed since the timestamp of the message
```

Comment:
note to myself: find out if this number is correct

Reply:
i dont think grade 4 is implemented @capossele ?

Reply:
it's probably not! 
I actually meant 
"grade 4 = something you can snapshot away"
but I did not want to mention snapshots